src/pages/workspaces/WorkspacesList/WorkspaceUserActions.ts

```diff
@@ -0,0 +1,24 @@
+import { Atom, atom } from '@terra-ui-packages/core-utils';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+/**
+ * This is here instead of libs/state.ts
+ * because it is limited to tracking current user activity in the List component
+ */
+export const workspaceUserActionsStore: Atom<WorkspaceUserActions> = atom<WorkspaceUserActions>({
```

Comment:
why atom and not useState<WorkspaceUserActions>?

also, I hope to have a useModalManager hook merged into dev soon that should make how we do this modal management less clunky, including sending in a modal args value/object on modal.open.

part of:

https://github.com/DataBiosphere/terra-ui/pull/4179

Reply:
Agreed that this looks like a case for component, not global, state.

Complex state might be worth a custom hook around `useState` or [`useReducer`](https://react.dev/reference/react/useReducer).

Context can also be used for "passing data deeply into the tree" if you're trying avoid props plumbing.
https://react.dev/reference/react/useContext#passing-data-deeply-into-the-tree

Reply:
That seemed to be the common way to create and access a shared store across components in this project.

Or do you mean 'why use a shared store at all'?
In which case, it's because in the original component, there are multiple places where the user can perform an action that will open a modal, and passing around numerous xxId/setXxId was cumbersome and hard to keep track of. 
It's plausible it could be solved by redesigning how the components are structured on a more fundamental level, but this seemed like the simpler and less error-prone solution.